src/main/java/org/opensearch/sdk/sample/helloworld/rest/RestHelloAction.java

```diff
@@ -39,31 +38,32 @@ public List<Route> routes() {
 
     @Override
     public ExtensionRestResponse handleRequest(ExtensionRestRequest request) {
-        // We need to track which parameters are consumed to pass back to OpenSearch
-        List<String> consumedParams = new ArrayList<>();
         Method method = request.method();
-        String uri = request.uri();
 
-        if (Method.GET.equals(method) && "/hello".equals(uri)) {
-            return new ExtensionRestResponse(OK, String.format(GREETING, worldName), consumedParams);
-        } else if (Method.PUT.equals(method) && uri.startsWith("/hello/")) {
-            // Placeholder code here for parameters in named wildcard paths
-            // Full implementation based on params() will be implemented as part of
-            // https://github.com/opensearch-project/opensearch-sdk-java/issues/111
-            String name = uri.substring("/hello/".length());
-            consumedParams.add("name");
-            try {
-                worldName = URLDecoder.decode(name, StandardCharsets.UTF_8);
-            } catch (IllegalArgumentException e) {
-                return new ExtensionRestResponse(BAD_REQUEST, e.getMessage(), consumedParams);
-            }
-            return new ExtensionRestResponse(OK, "Updated the world's name to " + worldName, consumedParams);
+        if (Method.GET.equals(method)) {
+            return handleGetRequest(request);
+        } else if (Method.PUT.equals(method)) {
+            return handlePutRequest(request);
         }
-        return new ExtensionRestResponse(
-            NOT_FOUND,
-            "Extension REST action improperly configured to handle " + method.name() + " " + uri,
-            consumedParams
-        );
+        return handleBadRequest(request);
+    }
+
+    private ExtensionRestResponse handleGetRequest(ExtensionRestRequest request) {
+        return new ExtensionRestResponse(request, OK, String.format(GREETING, worldName));
+    }
+
+    private ExtensionRestResponse handlePutRequest(ExtensionRestRequest request) {
+        String name = request.param("name");
+        try {
+            worldName = URLDecoder.decode(name, StandardCharsets.UTF_8);
+        } catch (IllegalArgumentException e) {
+            return new ExtensionRestResponse(request, BAD_REQUEST, e.getMessage());
+        }
+        return new ExtensionRestResponse(request, OK, "Updated the world's name to " + worldName);
+    }
+
```

Comment:
Can we have a sample POST request too here?

Reply:
Can I do that as part of #165?

Reply:
Sure @dbwiddis! Not blocking this PR :)